### PR TITLE
Refactoring classes to use static Make* construction functions

### PR DESCRIPTION
**Pull request by jankrassnigg**
Notable changes:

* MakeXYZ functions for the math classes and some others
* ezQuat doesn't use an ezVec3 internally anymore for the 'vector part'. So far the tests look good, but I am not 100% sure this can't create any problems with the editor data.

This PR only adds the new functions and marks up which functions would be deprecated in the future. It doesn't make any changes to the rest of the code yet, as that would be quite a large PR that's difficult to review, so that will come later.

### CODE REVIEW

**Review by C-Core — COMMENTED** [review]
I think this naming convention with the "Make" prefix is much better than the "s" prefix and more in line with the "ez style" of naming things. So green light from my side.

File: Code/Engine/Foundation/Time/Time.h [hunk 1]
```diff
@@ -15,33 +15,40 @@ struct EZ_FOUNDATION_DLL ezTime
   static ezTime Now(); // [tested]
 
   /// \brief Creates an instance of ezTime that was initialized from nanoseconds.
-  EZ_ALWAYS_INLINE constexpr static ezTime Nanoseconds(double fNanoseconds) { return ezTime(fNanoseconds * 0.000000001); }
+  [[nodiscard]] EZ_ALWAYS_INLINE constexpr static ezTime MakeNanoseconds(double fNanoseconds) { return ezTime(fNanoseconds * 0.000000001); }
```
Comment (C-Core) [inline Code/Engine/Foundation/Time/Time.h]: Shouldn't that be MakeFromNanoseconds etc. to follow the naming scheme of the other classes?

File: Code/UnitTests/FoundationTest/SimdMath/SimdTransformTest.cpp [hunk 2]
```diff
@@ -38,6 +38,29 @@ EZ_CREATE_SIMPLE_TEST(SimdMath, SimdTransform)
       EZ_TEST_BOOL((t.m_Scale == ezSimdVec4f(1)).AllSet<3>());
     }
 
+    {
+      ezSimdQuat qRot;
```
Comment (C-Core) [inline Code/UnitTests/FoundationTest/SimdMath/SimdTransformTest.cpp:42]: Use new MakeFromAxisAndAngle function

**Review by C-Core — APPROVED** [review]

File: Code/UnitTests/FoundationTest/SimdMath/SimdTransformTest.cpp (hunk 2 above)
Comment (jankrassnigg) [reply Code/UnitTests/FoundationTest/SimdMath/SimdTransformTest.cpp:42]: All these changes come in a separate PR.

File: Code/Engine/Foundation/Time/Time.h (hunk 1 above)
Comment (jankrassnigg) [reply Code/Engine/Foundation/Time/Time.h]: Theoretically yes, but that makes it even longer. I don't know, how do you feel about it ?

Comment (C-Core) [reply Code/Engine/Foundation/Time/Time.h]: I would prefer consistency in this case

Comment (jankrassnigg) [reply Code/Engine/Foundation/Time/Time.h]: Done

**Pull request merged**